src/Controllers/AnnouncementController.php

```diff
@@ -104,4 +107,78 @@ public function edit(Request $request, $id)
             return ApiResponse::failed($e);
         }
     }
+
+    public function delete($id)
+    {
+        try {
+            $user = auth()->user();
+
+            $announcement = Announcement::where('id', $id)
+                ->first();
+
+            if (! $announcement) {
+                throw ValidationException::withMessages([
+                    'id' => 'Announcement not found',
+                ]);
+            }
+
+            $authorized_announcement = Announcement::where('id', $id)
+                ->where('created_by', $user->id)
+                ->first();
+
+            if (! $authorized_announcement) {
+                $course = Course::where('id', $announcement->course_id)
+                    ->where('created_by', $user->id)
+                    ->first();
+
+                if (! $course) {
+                    throw ValidationException::withMessages([
+                        'id' => 'Announcement not found',
+                    ]);
+                }
+            }
+
+            if (! CourseUserHelper::isUserInCourse($user->id, $announcement->course_id)) {
+                throw ValidationException::withMessages([
+                    'id' => 'Must enroll the course to edit the announcement',
+                ]);
+            }
```

Comment:
Instead of separating the logic into 3 different parts, why don't we just merge it into 1 (or maybe 2).

- First, fetch the announcement with the corresponding id and user_id
  - If not found, then responds with not found
- Second, check if the user is enrolled to the course the announcement located in
  - If not enrolled, then responds with "Must enroll blalblabla"

This way, it'll be simpler and cheaper since we only need 2 database queries (instead of 4), 1 to get the announcement and 1 to check whether the user in the corresponding course (is called in the `CourseUserHelper::isUserInCourse`).